msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -930,6 +931,19 @@ public void onTaskCompleted(Boolean isSharedDevice) {
 
                         try {
                             if (config.getAccountMode() == AccountMode.SINGLE || isSharedDevice) {
+                                if (null != config.getDefaultAuthority() &&
+                                        config.getDefaultAuthority() instanceof AzureActiveDirectoryB2CAuthority) {
+                                    Logger.warn(
+                                            TAG,
+                                            "Warning! B2C applications should use MultipleAccountPublicClientApplication. "
+                                                    + "Use of SingleAccount mode with multiple IEF policies is unsupported."
+                                    );
+
+                                    if (config.getAuthorities().size() > 1) {
+                                        throw new MsalClientException(SAPCA_USE_WITH_MULTI_POLICY_B2C);
+                                    }
+                                }
+
```

Comment:
Could potentially do all this check earlier (prior to doing the GetDeviceModeCommand) and fail early if we are destined to fail

Reply:
Thanks @shahzaibj -- applied this feedback in c3e5a4bd8b4bd37e0c4a10bd03c8c747021fffdd